### PR TITLE
Reduce warnings in Emby.Dlna

### PR DESCRIPTION
**Pull request by Ullmie02**
#2149

Reduces the warnings in Emby.Dlna to 189

### CODE REVIEW

File: Emby.Dlna/Didl/StringWriterWithEncoding.cs [hunk 1]
```diff
@@ -20,7 +21,7 @@ public StringWriterWithEncoding(IFormatProvider formatProvider)
         }
 
         public StringWriterWithEncoding(StringBuilder sb)
-            : base(sb)
+            : base(sb, CultureInfo.InvariantCulture)
```
Comment (Bond-009) [inline Emby.Dlna/Didl/StringWriterWithEncoding.cs:24]: You should keep it as it was and disable this warning for this file to keep behavior consistent (and we should probably remove this file in the future)

File: Emby.Dlna/Main/DlnaEntryPoint.cs [hunk 2]
```diff
@@ -399,11 +401,36 @@ private void DisposePlayToManager()
...
         public void Dispose()
         {
-            DisposeDevicePublisher();
-            DisposePlayToManager();
-            DisposeDeviceDiscovery();
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
```
Comment (Bond-009) [inline Emby.Dlna/Main/DlnaEntryPoint.cs:421]: Add docs (just copy them from somewhere else)

File: Emby.Dlna/PlayTo/PlayToManager.cs [hunk 3]
```diff
@@ -247,8 +247,9 @@ public void Dispose()
             {
                 _disposeCancellationTokenSource.Cancel();
             }
-            catch
+            catch (Exception ex)
             {
+                _logger.LogDebug("Error while disposing PlayToManager", ex);
```
Comment (Bond-009) [inline Emby.Dlna/PlayTo/PlayToManager.cs:252]:
```suggestion
                _logger.LogDebug(ex, "Error while disposing PlayToManager");
```

File: Emby.Dlna/Service/BaseControlHandler.cs [hunk 4]
```diff
@@ -15,18 +15,18 @@ namespace Emby.Dlna.Service
 {
     public abstract class BaseControlHandler
     {
-        private const string NS_SOAPENV = "http://schemas.xmlsoap.org/soap/envelope/";
-
-        protected IServerConfigurationManager Config { get; }
-
-        protected ILogger Logger { get; }
+        private const string NsSoapenv = "http://schemas.xmlsoap.org/soap/envelope/";
```
Comment (Bond-009) [inline Emby.Dlna/Service/BaseControlHandler.cs:18]:
```suggestion
        private const string NsSoapEnv = "http://schemas.xmlsoap.org/soap/envelope/";
```

File: Emby.Dlna/Service/ControlErrorHandler.cs [hunk 5]
```diff
@@ -10,7 +10,7 @@ namespace Emby.Dlna.Service
 {
     public static class ControlErrorHandler
     {
-        private const string NS_SOAPENV = "http://schemas.xmlsoap.org/soap/envelope/";
+        private const string NsSoapenv = "http://schemas.xmlsoap.org/soap/envelope/";
```
Comment (Bond-009) [inline Emby.Dlna/Service/ControlErrorHandler.cs:13]:
```suggestion
        private const string NsSoapEnv = "http://schemas.xmlsoap.org/soap/envelope/";
```

**Review by Bond-009 — CHANGES_REQUESTED** [review]

File: Emby.Dlna/Main/DlnaEntryPoint.cs (hunk 2 above)
Comment (Ullmie02) [reply Emby.Dlna/Main/DlnaEntryPoint.cs:425]: Added docs to two other `Dispose `methods in Emby.Dlna as well

**Comment by Ullmie02** [comment]
Don't know why CI is failing. Local build works fine. https://dev.azure.com/jellyfin-project/jellyfin/_build/results?buildId=13031&view=results seems to be failing because of the same error

**Review by Bond-009 — APPROVED** [review]

**Review by barronpm — APPROVED** [review]

**Pull request merged**